recipes/szip/all/conanfile.py

```diff
@@ -102,17 +102,25 @@ def _create_cmake_module_alias_targets(module_file, targets):
                     set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
                 endif()
             """.format(alias=alias, aliased=aliased))
-        tools.save(module_file, content)
+        save(self, module_file, content)
 
     @property
     def _module_file_rel_path(self):
-        return os.path.join("lib", "cmake", "conan-official-{}-targets.cmake".format(self.name))
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
 
     def package_info(self):
-        self.cpp_info.set_property("cmake_file_name", "szip")
+        self.cpp_info.set_property("cmake_file_name", "SZIP")
         self.cpp_info.set_property("cmake_target_name", "szip-shared" if self.options.shared else "szip-static")
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.libs = collect_libs(self)
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.extend(["m"])
+
+        if self.options.shared:
+            self.cpp_info.defines.append("SZ_BUILT_AS_DYNAMIC_LIB=1")
 
         # TODO: to remove in conan v2 once cmake_find_package* generators removed
+        self.cpp_info.names["cmake_find_package"] = "SZIP"
+        self.cpp_info.names["cmake_find_package_multi"] = "SZIP"
```

Comment:
```suggestion
```

Reply:
I thought we still need these lines?  It is in the v2 package_template in the docs folder...